src/options.cpp

```diff
@@ -257,6 +257,21 @@ int options_parse(int argc, char **argv, lmp_options &opt)
 	if (opt.uuid.empty())
 		get_uuid(opt);
 
+	/* Validate the UUID early to avoid HTTP 400 */
+	std::regex UUID("^"
+		"[a-fA-F0-9]{8}-"
+		"[a-fA-F0-9]{4}-"
+		"[a-fA-F0-9]{4}-"
+		"[a-fA-F0-9]{4}-"
+		"[a-fA-F0-9]{12}"
+		"$");
+	std::smatch match;
+	std::regex_search(opt.uuid, match, UUID);
+	if (match.size() <= 0) {
+		cerr < "Invalid UUID: " << opt.uuid << endl;
+		return -1
+	}
```

Comment:
We don't require this thing to be a valid UUID.
In fact, we have customers which use a non-UUID string here.

What the server expects is that this field only contains `[a-zA-Z0-9_\-\.]` charaters.
It'd be wise to also have some length limit, e.g. 60 characters.
No more than that is needed.

Reply:
I guess it would be more appropriate if we then fix the server.

The option is very clear: 
```
#define UUID_HELP \
"A per-device UUID. If not provided, one will be generated. This is "          \
"associated with the device, e.g. as the CommonName field in certificates "    \
"related to it."

```

This not being a new option seems to indicate that users might have been misled by our lack of validation both in the client and in the server side; and this has allowed them to and push weird strings.

IMO we should  make it right, merge this PR and fix the server to do the same thing.

Reply:
you should check and make sure el2g produces "valid" UUIDs. I think the format they use is different and might be why we have the rules in place we do.

Reply:
el2g does not generate UUIDs.

What I see is that when the user does not provide a UUID on the input command line, either we use openssl to generate one OR we parse the UUID from the pkcs#11 token.

I guess you point is that maybe OP-TEE could be generating a different UUID format? but if that is the case we should perhaps parse differently.

or we could completely ignore the PR and keep going as we do now and let the server validate in any way it wants to validate.

Reply:
unless there is some EL2G service that generates a UUID and then calls lmp-device-register command line?

Reply:
el2g is its own service that doesn't use lmp-device-register. Devices created with el2g have "nxp-*" prefixes for their UUIDs.

Reply:
el2g wouldn't use lmp-device-register, so it should be at least ok to have this validation done here at least (as the tool is asking the user to use a valid uuid).

Reply:
ah I remember now. Well in the same way that I modified the el2go registration service to support the OP-TEE tunnel instead of the i2c bus, we could modify it to generate only valid UUIDs.

but maybe asking for consistency on the client and server side is asking too much for little gain.

Reply:
i don't have a big opinion. el2g wont touch this client or server side logic you mention. So maybe we make things more strict for now - then if someone comes around and needs it *less* strict, we can make a more informed decision.

Reply:
Yeah, let's enforce this here only for now, as making this interface flexible will generate unexpected results.

Reply:
ok to merge then?

Reply:
LGTM